### ISSUE TITLE
Pitch and speed dialog: names read by screen readers are incorrect

### ISSUE THREAD

**Issue by DavidBailes** [audacity/audacity#5912]
### Steps to reproduce

On Windows 10, with a screen reader running:
1. Select an audio clip
2. Open the Pitch and speed dialog.
3. Tab round the controls in the dialog.

### Expected behavior

The names of the first three controls should be:
Clip pitch semitones
Clip pitch cents
Clip speed %

### Actual behavior

The names of the first three controls are:
Clip pitch
semitones,
%

### Audacity Version

current master / alpha / nightly build

### Operating system

Windows

**Comment by DavidBailes** [comment]
@saintmatthieu

The fix for the third control is straightforward change the line:
`auto txtCtrl = s.NameSuffix(Verbatim("%"))`
to
`auto txtCtrl = s.Name(XO("Clip Speed")).NameSuffix(Verbatim("%"))`

However the fix for the first two controls, the two spin boxes is not as straightforward.
If the desired accessibility name is just the text immediately before the control, there isn't a problem - windows automatically sets the accessibility name to this text. But if it isn't, which is the case here, things are complicated by the fact that the focus is a text box within the control. In the Time signature toolbar there is a similar issue, and @crib created a custom SpinControl to get round this problem.

**Comment by saintmatthieu** [comment]
Hi @DavidBailes, thanks for your report.
It's interesting, now I think I understand better the meaning of some of the ShuttleGui methods.
I could improve it easily by using the `TieSpinCtrl` method like one is supposed to, but then "semitones" and "cents" would come before the boxes instead: 
![image](https://github.com/audacity/audacity/assets/22740106/5a58a18a-a1fe-4eea-885d-ac0dcf5ebbb6)
instead of currently:
![image](https://github.com/audacity/audacity/assets/22740106/21527b9a-ee99-4c68-a8b1-a2fe768779f5)

I don't know what @LWinterberg would think of that, the current layout was his specification.

**Comment by LWinterberg** [comment]
using a group box around the pitch controls seems fine for the time being

**Comment by LWinterberg** [comment]
![image](https://github.com/audacity/audacity/assets/87814144/4f5dc90a-1d82-4455-8d24-343a5ce3e4dd)

so this basically

**Comment by DavidBailes** [comment]
> ![image](https://private-user-images.githubusercontent.com/87814144/302937938-4f5dc90a-1d82-4455-8d24-343a5ce3e4dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDczMDA5MTAsIm5iZiI6MTcwNzMwMDYxMCwicGF0aCI6Ii84NzgxNDE0NC8zMDI5Mzc5MzgtNGY1ZGM5MGEtMWQ4Mi00NDU1LThkMjQtMzQzYTVjZTNlNGRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA3VDEwMTAxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmMzg4ZDNhZDU0YmQ2MzUyZjFiZDkzODIwMjc2MjRjMjI4N2JmM2NhMTE1NzM5NDdhMWQxNWU1MWM1OGZmMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PXzStsWEQOVn8KcHwF7O1cl0O56c3opJJqeSJsK0EF4)
> 
> so this basically

You have changed the control for the Clip speed from a text box to a spin control. It will be very difficult for the accessibility name of that control to include the "%" if a wxSpinCtrl is used.
Please could the control for the Clip speed remain a text box?

**Comment by LWinterberg** [comment]
Clip speed may remain a text box.

**Comment by LWinterberg** [comment]
Yes

**Comment by DavidBailes** [comment]
@LWinterberg . Please can this be moved to "ready for review"?
